src/api/profile.api.js

```diff
@@ -1,80 +1,273 @@
 import { API_SERVICE_ALTI_BASE_URL } from '@/config'
+import { CoordinateSystems } from '@/utils/coordinateUtils'
 import log from '@/utils/logging'
 import axios from 'axios'
-
-/**
- * @typedef ProfileRequestData
- * @property {GeoJSON} geom A GeoJSON object representation of a polyline (type = LineString)
- * @property {Number} [sr] The reference system to use (EPSG code). Valid value are 2056 (for LV95)
- *   and 21781 (for LV03)
- * @property {Number} [nb_points] The number of points used for the polyline segmentation. Default
- *   “200”.
- * @property {Number} [offset] The offset value (INTEGER) in order to use the exponential moving
- *   algorithm. For a given value the offset value specify the number of values before and after
- *   used to calculate the average.
- * @property {Boolean} [distinct_points] If True, it will ensure the coordinates given to the
- *   service are part of the response. Possible values are True or False, default to False.
- */
+import { LineString } from 'ol/geom'
 
 /**
  * @typedef GeoJSON A GeoJSON object
  * @property {string} type
  * @property {[[Number]]} coordinates
  */
 
-/**
- * @typedef ProfileAlts
- * @property {Number} DTM2
- * @property {Number} COMB
- * @property {Number} DTM25
- */
+export class ProfilePoint {
+    /**
+     * @param {Number} dist Distance from first to current point
+     * @param {[Number, Number]} coordinate Coordinate of this point in EPSG:3857
+     * @param {Number} elevation In the COMB elevation model
+     */
+    constructor(dist, coordinate, elevation) {
+        this._dist = dist
+        this._coordinate = coordinate
+        this._elevation = elevation
+    }
 
-/**
- * @typedef ProfilePoint
- * @property {Number} dist Distance from first to current point
- * @property {Number} easting Coordinate
- * @property {Number} northing Coordinate
- * @property {ProfileAlts} alts Point altitude in different elevation models
- */
+    get dist() {
+        return this._dist
+    }
+
+    get coordinate() {
+        return this._coordinate
+    }
+
+    /** @returns {Number} The COMB elevation */
+    get elevation() {
+        return this._elevation
+    }
+
+    // no setter
+}
+
+export class GeoAdminProfile {
```

Comment:
I don't like naming things with only one word, like `Profile` as it is too generic. And I don't really like `Alti` too, because it doesn't mean anything in English... So I thought of this name as a "good enough" but am open to suggestions

Reply:
I totally agree with you `Profile` is a bad name as it is too general. To be honest I find `AltiProfile` a good choice as `Alti` is a shorthand for `Altitude` and it is an altitude profile. But here other ideas
- ElevationProfile
- GeoProfile